pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -523,12 +528,21 @@ func (p *PriorityQueue) AssignedPodUpdated(pod *v1.Pod) {
 // This function adds all pods and then signals the condition variable to ensure that
 // if Pop() is waiting for an item, it receives it after all the pods are in the
 // queue and the head is the highest priority pod.
-func (p *PriorityQueue) MoveAllToActiveOrBackoffQueue(event string) {
+func (p *PriorityQueue) MoveAllToActiveOrBackoffQueue(event string, fns ...PreEnqueueCheck) {
 	p.lock.Lock()
 	defer p.lock.Unlock()
 	unschedulablePods := make([]*framework.QueuedPodInfo, 0, len(p.unschedulableQ.podInfoMap))
 	for _, pInfo := range p.unschedulableQ.podInfoMap {
-		unschedulablePods = append(unschedulablePods, pInfo)
+		preCheckPassed := true
+		for _, fn := range fns {
```

Comment:
do we really need to have the interface support multiple functions? The caller can always wrap everything they want to check in a single function, and this logic here would be simplified to a couple of lines.

Reply:
> do we really need to have the interface support multiple functions?

We don't need to, it's just variadic functions can keep logic of existing callers intact.

Updated the code to change the signature to be `MoveAllToActiveOrBackoffQueue(event string, preCheckFn)` .